lib/ansible/modules/cloud/azure/azure_rm_resourcegroup_facts.py

```diff
@@ -148,6 +164,19 @@ def list_items(self):
                 results.append(self.serialize_obj(item, AZURE_OBJECT_CLASS))
         return results
 
+    def list_by_rg(self, name):
+        self.log('List resources under resource group')
+        results = []
+        try:
+            response = self.rm_client.resources.list_by_resource_group(name)
+            while True:
+                results.append(response.next().as_dict())
+        except StopIteration:
```

Comment:
why do you need to use StopIteration etc?
I think for item in response will handle the iteration including paged result.

Reply:
The for can only iterate the current page with limited size

Reply:
so if stopiteration that means this page is  done, but there might be other page?

Reply:
Here uses the .next() to query all items in the pages. If use `for`, the code may like
```python
[item.as_dict() for item in response.current_page]
```

Reply:
Well, "for" handles StopIteration exception and Next internally, so you can just have for loop